### PR TITLE
DRA: Handle env variables better

### PR DESCRIPTION
**Pull request by roaksoax**
## What does this PR do?

Introduce a new way to handle DRA WORKFLOW_TYPE environment variable.

NOTE: this will require the Jenkins definitions to set WORKFLOW_TYPE=standalone or WORKFLOW_TYPE=staging

### CODE REVIEW

**Review by andsel — CHANGES_REQUESTED** [review]
I like this refactoring but has an issue in generating the `STACK_VERSION`, plus I would switch the `staging` and `snapshot` paths so that the diff becomes more evident

File: ci/dra_aarch64.sh [hunk 1]
```diff
@@ -13,40 +13,45 @@ STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:[[:space:]]\([[:digit:]]*
...
+            rake artifact:docker_oss
+            rake artifact:dockerfiles
+        else
+            VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" rake artifact:docker
+            VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" rake artifact:docker_oss
+            VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" rake artifact:dockerfiles
+            # Qualifier is passed from CI as optional field and specify the version postfix
+            # in case of alpha or beta releases:
+            # e.g: 8.0.0-alpha1
+            STACK_VERSION="${STACK_VERSION}-${VERSION_QUALIFIER_OPT}"
```
[the remark this answers is not in the document]
Comment (roaksoax) [reply ci/dra_aarch64.sh:33]: This is copy / paste from the code you had, I simply reorganized it. Are you saying this is buggy?

File: ci/dra_x86_64.sh [hunk 2]
```diff
@@ -11,34 +11,37 @@ export JRUBY_OPTS="-J-Xmx1g"
 # The suffix part like alpha1 etc is managed by the optional VERSION_QUALIFIER_OPT environment variable
 STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:[[:space:]]\([[:digit:]]*\.[[:digit:]]*\.[[:digit:]]*\)$/\1/p'`
 
-# WORKFLOW_TYPE is a CI externally configured environment variable where, if not an empty string,
-# it's assumed to be snapshot
-if [ -n "$WORKFLOW_TYPE" ]; then
-  STACK_VERSION=${STACK_VERSION}-SNAPSHOT
-fi
+case "$WORKFLOW_TYPE" in
+    snapshot)
```
[the remark this answers is not in the document]
Comment (roaksoax) [reply ci/dra_x86_64.sh:15]: Same, dont understand what you mean.

File: ci/dra_aarch64.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (andsel) [reply ci/dra_aarch64.sh:33]: The bug is already present in the `main` branch, already existed

File: ci/dra_aarch64.sh [hunk 3]
```diff
@@ -13,40 +13,45 @@ STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:[[:space:]]\([[:digit:]]*
...
-    VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" rake artifact:dockerfiles
-    # Qualifier is passed from CI as optional field and specify the version postfix
-    # in case of alpha or beta releases:
-    # e.g: 8.0.0-alpha1
-    STACK_VERSION="${STACK_VERSION}-${VERSION_QUALIFIER_OPT}"
-  fi
-fi
+
+case "$WORKFLOW_TYPE" in
+    snapshot)
```
[the remark this answers is not in the document]
Comment (andsel) [reply ci/dra_aarch64.sh:18]: doesn't mind

**Review by andsel — APPROVED** [review]
LGTM

**Comment by roaksoax** [comment]
Failures due to an unrelated error. Landing this.

**Comment by roaksoax** [comment]
@logstashmachine backport 8.5

**Comment by roaksoax** [comment]
@logstashmachine backport 7.17

**Pull request merged**